Added details on wildcards in `pkg`

stellarpower opened the pull request:
Added a short description of how globbing is used in `pkg`, e.g. in `pkg list -a` to search for software. Please feel free to edit if there are inaccuracies or changes needed, mainly I just wanted to add something as it's useful to know!

citrus-it commented on _docs/info/getstarted.md:
````diff
@@ -92,7 +92,35 @@ developer/gcc5                                    5.5.0-0.151024             ---
 ```
 
-and then install the desired version along with the `system/header` package:
+Note that our query is run against the **entire package name**, including its category (in this case, developer), however a short form can be used. For example, `bhyve` would match against `system/bhyve`, `system/bhyve/firmware`, `system/library/bhyve`, and `system/zones/brand/bhyve`, but `gcc` would not match against `gcc5` due to the extra `5`. Globbing is provided (remember to use quotes if us ...
````
The short form only matches the trailing part, so `bhyve` does not match `system/bhyve/firmware`, although `firmware` would and so would `bhyve/firmware`.

I think you can drop the `if using e.g. bash` bit - if the command is being used interactively, then the quotes are going to be necessary regardless of what shell is in use.

stellarpower replied:
I actually ran that query for `bhyve` and copied the output in - I'll run it again and double-check and then edit.

citrus-it commented on _docs/info/getstarted.md:
```diff
@@ -92,7 +92,35 @@ developer/gcc5                                    5.5.0-0.151024             ---
+sfe/system/library/gcc-49-runtime (localhostomnios) 4.9.4-0.151030.3           ---
+sfe/system/library/gcc-54-runtime (localhostomnios) 5.4.0-0.0.151014           ---
+sfe/system/library/gcc-7-runtime (localhostomnios) 7.3.0-0.0.151022           ---
+sfe/system/library/gcc-runtime (localhostomnios)  4.9.4-0.151030.3           ---
```
Please remove the sfe ones from this example list.

stellarpower replied:
I'll think of a different example - it was late and gcc was what I had searched for, so I was trying to think of an example where a version number or some other suffix that we might not be expecting might prevent us from finding the package we'd be expecting, esp. if coming from other OSs. However the number of packages for gcc in the core repo was somewhat low, so I'll fig around for a better example.

citrus-it commented on _docs/info/getstarted.md:
````diff
@@ -92,7 +92,35 @@ developer/gcc5                                    5.5.0-0.151024             ---
+```
+
+
+We can then install the desired version along with the `system/header` package:
````
Since this was initially written, the `build-essential` package was added so generally if you want to build software that's what we recommend installing now. I don't know if you can work this in instead of the existing `along with the system/header package`.

```
% pkgtree -Ul depends build-essential

|------(    require)--pkg://omnios/developer/build-essential@11-151032.0:20191028T102056Z
| pkg://omnios/system/library/math
| pkg://omnios/developer/build/make
| pkg://omnios/developer/gcc7
| pkg://omnios/developer/build/autoconf
| pkg://omnios/developer/object-file
| pkg://omnios/developer/build/automake
| pkg://omnios/developer/parser/bison
| pkg://omnios/developer/build/gnu-make
| pkg://omnios/developer/lexer/flex
```

stellarpower replied:
As I was discussing writing a readme for building packages, should I perhaps incorporate this into that, i.e. a guide for building software on Omni, and edit this page to reflect a more general package installation, that still gives a useful example for mediators and the globs further up?

citrus-it commented:
@stellarpower Do you have any more updates for this PR?

stellarpower commented:
Sorry, not yet, haven't had the time I'm afraid however it's on my mental todo list. I really want to get an installation running on real hardware right now because trying to run Omni as a guest itself under linux is causing so many extra headaches but I've not been successful in that thus far.

oetiker commented:
ping

stellarpower commented:
I've done some reworking and hopefully addressed the issues in the first version, please let me know what you think! I've separated package queries into a new section and perhaps have been a bit verbose but I thought best to be overly rather than underly specific. I've also moved the build-essentials in with the installation steps as an alternative option.
